components/support/base/src/main/java/mozilla/components/support/base/ids/SharedIds.kt

```diff
@@ -48,9 +49,29 @@ internal class SharedIds(
         editor.putInt(KEY_NEXT_ID, nextId + 1) // Ignoring overflow for now.
         editor.putInt(key, nextId)
         editor.putLong(lastUsedKey, now())
+        editor.apply()
+
+        return nextId
+    }
+
+    /**
+     * Get the next available unique ID for the provided unique tag.
+     */
+    @Synchronized
+    fun getNextIdForTag(context: Context, tag: String): Int {
+        val preferences = preferences(context)
+        val editor = preferences.edit()
+        val key = tagToKey(tag)
+        val lastUsedKey = tagToLastUsedKey(tag)
 
         removeExpiredIds(preferences, editor)
 
+        // always use the next available one and save that.
```

Comment:
Should we update the `lastUsedKey` when we last access this as well (just not return).

Reply:
`lastUsedKey` will be updated below on line 74.  Since `lastUsedKey` is the same when the tag is the same, it will be overwritten.

Reply:
Ah yes, I missed that!